src/GitTools.jl

```diff
@@ -66,35 +66,24 @@ function transfer_progress(progress::Ptr{LibGit2.TransferProgress}, p::Any)
 end
 
 const GIT_REGEX =
-    r"^(?:(?:git|ssh|https)(?:://|@))?(?:[\w\.\+\-:]+@)?([^/:]+)[:/](.+?)(?:\.git)?$"i
-const GITHUB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
-const GITLAB_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
-const FALLBACK_PROTOCOL = Ref{Union{String, Nothing}}(nothing)
+    r"^(?:(?<proto>git|ssh|https)://)?(?:[\w\.\+\-:]+@)?(?<hostname>.+?)(?(<proto>)/|:)(?<path>.+?)(?:\.git)?$"
+const GIT_PROTOCOL = Dict{AbstractString, Union{Nothing, AbstractString}}()
```

Comment:
Just go with `String` instead of `AbstractString` here. The `Dict` data structure should convert for you.